### PR TITLE
【Prim】Fix prim amp

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

Bug fixes
### PR changes

Others
### Describe

Fix error in amp mode, incompletely support amp in batch_norm

### CODE REVIEW

File: python/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -1889,7 +1889,7 @@ def linear(x, weight, bias=None, name=None):
                 type='elementwise_add',
                 inputs={'X': [tmp], 'Y': [bias]},
                 outputs={'Out': [res]},
-                attrs={'axis': len(x.shape) - 1},
+                attrs={'axis': -1},
```
Comment (xiaoguoguo626807) [inline python/paddle/nn/functional/common.py:1892]: why only support axis = -1

**Review by xiaoguoguo626807 — APPROVED** [review]

File: python/paddle/nn/functional/common.py (hunk 1 above)
Comment (JiabinYang) [reply python/paddle/nn/functional/common.py:1892]: Because in phi standard we only allow axis=-1

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_batch_norm.py [hunk 2]
```diff
@@ -258,5 +262,69 @@ def test_forward(self):
...
+        with paddle.amp.auto_cast(level='O2'):
+            out = net(self.x)
+            loss = paddle.mean(out)
+            loss.backward()
+            sgd.step()
+            sgd.clear_grad()
+            return loss
+
+    def test_amp(self):
+        if not isinstance(framework._current_expected_place(), core.CPUPlace):
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_batch_norm.py:318]: check place because cpu does not support float16?

File: python/paddle/incubate/autograd/composite_rules.py [hunk 3]
```diff
@@ -59,6 +59,13 @@ def composite_batchnorm(
     trainable_statistics,
 ):
     """define composite rule of op batch_norm"""
+    is_amp = False
+    from paddle.fluid.data_feeder import convert_dtype
+
+    if convert_dtype(x.dtype) == "float16":
+        print("Running batch_norm in amp")
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:66]: better to add comment to explain we do this

File: python/paddle/incubate/autograd/composite_rules.py [hunk 4]
```diff
@@ -59,6 +59,13 @@ def composite_batchnorm(
     trainable_statistics,
 ):
     """define composite rule of op batch_norm"""
+    is_amp = False
+    from paddle.fluid.data_feeder import convert_dtype
+
+    if convert_dtype(x.dtype) == "float16":
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:65]: if not in amp case but x.shape = "float16", should we also do this?

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_batch_norm.py (hunk 2 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_batch_norm.py:318]: Because cinn doesn't support cpu

File: python/paddle/incubate/autograd/composite_rules.py (hunk 4 above)
Comment (JiabinYang) [reply python/paddle/incubate/autograd/composite_rules.py:65]: May not have this case, I think

File: python/paddle/incubate/autograd/composite_rules.py (hunk 3 above)
Comment (JiabinYang) [reply python/paddle/incubate/autograd/composite_rules.py:66]: sure,but how about in next pr when we support autocast strategy

**Review by cyber-pioneer — APPROVED** [review]

**Comment by cyber-pioneer** [comment]
LGTM
